redhat-developer/service-binding-operator
# Split BindingReady into CollectionReady and InjectionReady

### Motivation

To fix  #381 

### Changes

Splits the `BindingReady` type into more specific types:
- CollectionReady which would be related to the collection and persistence of intermediate manifests (secrets, config maps, etc)
- InjectionReady which would be related to change the applications' manifests to use those intermediate manifests:

### How to test

1. Follow the steps given in this [example](https://github.com/redhat-developer/service-binding-operator/tree/master/examples/nodejs_postgresql) to:
- install DB operator, Service Binding Operator
- Import an application
- Create a DB instance 

2. Create the following `ServiceBindingRequest`:

```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  name: binding-request
  namespace: service-binding-demo
spec:
 backingServiceSelector:
    group: postgresql.baiju.dev
    version: v1alpha1
    kind: Database
    resourceRef: db-demo
```

3. Use the following command to check if the status has been updated:
`oc get ServiceBindingRequest binding-request -o yaml`

The output should have `status.conditions.type` as 
- `CollectionReady` with `status: "True"`
-  `InjectionReady` with `status: "False"`

```
status:
  conditions:
  - lastHeartbeatTime: 2020-04-23T06:25:16Z
    lastTransitionTime: 2020-04-23T06:25:16Z
    message: application ResourceRef or MatchLabel not found
    reason: EmptyApplicationSelector
    status: "True"
    type: CollectionReady
  - lastHeartbeatTime: 2020-04-23T06:25:16Z
    lastTransitionTime: 2020-04-23T06:25:16Z
    message: application ResourceRef or MatchLabel not found
    reason: EmptyApplicationSelector
    status: "False"
    type: InjectionReady
```
